### ISSUE TITLE
Provide a way to get all the extract emoji by aliases

### ISSUE THREAD

**Issue by freya022** [felldo/JEmoji#11]
### What feature are you proposing?

As far as I can tell, `EmojiManager#extractEmojisInOrder` only extract unicode emojis.

I think what should be added is either a method that extract emojis by their aliases, or `EmojiManager#extractEmojisInOrder` should extract both unicode and aliases

**Comment by felldo** [comment]
Hi, this is not as easy as it sounds because it depends on the actual platform you want to use the alias from. For example `:call_me_hand::skin-tone-2:` discord an slack both have this alias for 🤙 with a tone. But both platforms treat them different and display a different tone for this kind of emoji. Which one should be used in this case? So I don't think there's really a reliable way to do this without having a dedicated method that tries to extract the emojis by alias for the specific platform.

> I would also like if it was possible to get the index of the extracted emojis.

Getting the index of something is a pretty broad term. Internally JEmoji uses an array of codepoints and iterates over it to determine whether something is an emoji. You probably want to get the index of the String where the emoji appears?
Regardless of that, I think that's a good addition with the only (probably) diadvantage of performance loss due to trying to find the index in the String. I will add `EmojiManager#extractEmojisInOrderWithIndex` to my to do list 🙂

**Comment by freya022** [comment]
> Getting the index of something is a pretty broad term. Internally JEmoji uses an array of codepoints and iterates over it to determine whether something is an emoji. You probably want to get the index of the String where the emoji appears? Regardless of that, I think that's a good addition with the only (probably) diadvantage of performance loss due to trying to find the index in the String.

Yes, I'd want the index of the String where the emoji appears, if I read the source correctly, it should be the [textIndex](https://github.com/felldo/JEmoji/blob/badbc4e2d885511de308ea25aca4d109c554a715/lib/src/main/java/net/fellbaum/jemoji/EmojiManager.java#L284) variable

**Comment by freya022** [comment]
I'll do a PR, I have some idea on how to refactor this
